src/aliceVision/sfmData/colorize.cpp

```diff
@@ -31,97 +62,52 @@ void colorizeTracks(SfMData& sfmData)
   for(auto& landmarkPair : sfmData.getLandmarks())
     remainingLandmarksToColor.push_back(landmarkPair.second);
 
-  struct ViewInfo
-  {
-    ViewInfo(IndexT viewId, std::size_t cardinal)
-      : viewId(viewId)
-      , cardinal(cardinal)
-    {}
-
-    IndexT viewId;
-    std::size_t cardinal;
-    std::vector<std::reference_wrapper<Landmark>> landmarks;
-  };
-
-  std::vector<ViewInfo> sortedViewsCardinal;
-  sortedViewsCardinal.reserve(sfmData.getViews().size());
-  {
-    // create cardinal per viewId map
-    std::map<IndexT, std::size_t> viewsCardinalMap; // <ViewId, Cardinal>
-    for(const auto& landmarkPair : sfmData.getLandmarks())
-    {
-      const Observations& observations = landmarkPair.second.observations;
-      for(const auto& observationPair : observations)
-        ++viewsCardinalMap[observationPair.first]; // TODO: 0
-    }
-
-    // copy key-value pairs from the map to the vector
-    for(const auto& cardinalPair : viewsCardinalMap)
-      sortedViewsCardinal.push_back(ViewInfo(cardinalPair.first, cardinalPair.second));
-
-    // sort the vector, biggest cardinality first
-    std::sort(sortedViewsCardinal.begin(),
-              sortedViewsCardinal.end(),
-              [] (const ViewInfo& l, const ViewInfo& r) { return l.cardinal > r.cardinal; });
-  }
+  std::vector<LMColorAccumulator> landmarkInfo(sfmData.getLandmarks().size());
+  const Views& views = sfmData.getViews();
 
-  // assign each landmark to a view
-  for(ViewInfo& viewCardinal : sortedViewsCardinal)
+  #pragma omp parallel for
```

Comment:
enable nested and limit io to 3 threads
```suggestion
    omp_set_nested(1);
    #pragma omp parallel for num_threads(3)
```

Reply:
This results in slower colorization. Would you like to increase num_threads to 5? I guess num_threads limit depends on processing unit in the user's system